src/server/search/search_family.cc

```diff
@@ -204,15 +208,26 @@ std::string_view ParseFieldWithAtSign(CmdArgParser* parser) {
 }
 
 void ParseLoadFields(CmdArgParser* parser, std::optional<OwnedSearchFieldsList>* load_fields) {
+  // TODO: Change to num_strings. In Redis strings number is expected. For example: LOAD 3 $.a AS a
   size_t num_fields = parser->Next<size_t>();
   if (!load_fields->has_value()) {
     load_fields->emplace();
   }
 
   while (num_fields--) {
-    string_view field = ParseField(parser);
-    string_view alias = parser->Check("AS") ? parser->Next() : field;
-    load_fields->value().emplace_back(field, alias);
+    string_view str = parser->Next();
+
+    if (StartsWithAtSign(str)) {
+      str.remove_prefix(1);  // remove leading @
+    }
+
+    StringOrView name = StringOrView::FromString(std::string{str});
+    if (parser->Check("AS")) {
+      load_fields->value().emplace_back(name, true,
```

Comment:
I see you add aliases as strings. Where do you benefit from string_view, in other words, why using `StringOrView` types?